crates/holochain/tests/tests/countersigning/session_interaction_over_websocket.rs

```diff
@@ -0,0 +1,798 @@
+            &[alice_response.clone(), bob_response.clone()],
+        )
+        .await;
+
+        if let AppResponse::ZomeCalled(_) = response {
+            break;
+        }
+        tokio::time::sleep(Duration::from_millis(100)).await;
+    }
+
+    tracing::info!("Alice committed countersigned entry - restarting conductor to provoke unresolvable state...\n");
+
+    // Restart Alice's conductor to put countersigning session in Unresolvable state.
+    let mut alice = restart_conductor(alice).await;
+
+    // Attach app interface to Alice's conductor.
+    let (alice_app_tx, mut alice_app_rx) = connect_app_interface(&mut alice).await;
+    // Spawn task listening to app socket messages, preventing app socket to be dropped.
+    tokio::spawn(async move { while alice_app_rx.recv::<AppResponse>().await.is_ok() {} });
+
+    // Alice's session should be in state Unresolvable with 1 attempted resolution.
+    tokio::time::timeout(Duration::from_secs(5), async {
+        loop {
+            let state = get_session_state(&alice.cell_id, &alice_app_tx).await;
+            if let Some(CountersigningSessionState::Unknown {
+                resolution: summary,
+                ..
+            }) = state
+            {
+                if summary.attempts == 1 {
+                    break;
+                }
+            }
+            tokio::time::sleep(Duration::from_millis(100)).await;
+        }
+    })
+    .await
+    .unwrap();
+
+    // Bob's session should still be in Accepted state.
+    assert_matches!(
+        get_session_state(&bob.cell_id, &bob_app_tx).await,
+        Some(CountersigningSessionState::Accepted(_))
+    );
+
+    // Alice abandons the session.
+    let response: AppResponse = request(
+        AppRequest::AbandonCountersigningSession(Box::new(alice.cell_id.clone())),
+        &alice_app_tx,
+    )
+    .await;
+    assert_matches!(response, AppResponse::CountersigningSessionAbandoned);
+
+    // Session should be abandoned and can not be abandoned again.
+    let response: AppResponse = request(
+        AppRequest::AbandonCountersigningSession(Box::new(alice.cell_id.clone())),
+        &alice_app_tx,
+    )
+    .await;
+    let expected_error = AppResponse::Error(
+        ConductorApiError::ConductorError(ConductorError::CountersigningError(
+            CountersigningError::SessionNotFound(alice.cell_id.clone()),
+        ))
+        .into(),
+    );
+    assert_eq!(format!("{:?}", response), format!("{:?}", expected_error));
+
+    // Alice's session should be gone.
+    assert_matches!(get_session_state(&alice.cell_id, &alice_app_tx).await, None);
+    // Bob's session should still be in Accepted state.
+    assert_matches!(
+        get_session_state(&bob.cell_id, &bob_app_tx).await,
+        Some(CountersigningSessionState::Accepted(_))
+    );
+
+    tracing::info!("Alice abandoned session.\n");
+
+    // Await Bob's session to be abandoned due to timeout.
```

Comment:
Does this still work? I guess it does. Holochain will make an automated retry, but unless you configure more retries it will just be the one and if that doesn't yield an outcome, the session will be abandoned.

Reply:
I don't understand. The session will be abandoned and the resulting signal is listened to here. Holochain will make an automated retry to send the signal?